shiny/_app.py

```diff
@@ -138,6 +140,15 @@ def _server(inputs: Inputs, outputs: Outputs, session: Session):
 
         self.starlette_app = starlette_app
 
+        if isfunction(ui):
```

Comment:
This won't catch non-function callable objects. I think you can use `callable()` instead: https://docs.python.org/3/library/functions.html#callable

(And same below on line 280)